Unit test tsc failure: error TS2306 index.d.ts is not a module

I am pretty stumped with this error and would like to see if I can solicit some help here.

Recently, the unit test failed when we try to run `tsc` over the grpc-web generated stub class. Here's the error:

```
~/grpc-web/packages/grpc-web$ ./node_modules/.bin/tsc ./test/echo_grpc_web_pb.d.ts                                                                                                             
node_modules/grpc-web.d.ts:1:15 - error TS2306: File '/usr/local/google/home/stanleycheung/grpc-web/packages/grpc-web/index.d.ts' is not a module.

1 export * from '/usr/local/google/home/stanleycheung/grpc-web/packages/grpc-web';                            
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                   
Found 1 error.
```

To reproduce:
- Clone this repo
- cd grpc-web
- sudo make install-plugin
- cd packages/grpc-web
- npm install
- npm test

In order to leave the generated code file not to be deleted, you can comment out this line:

```
diff --git a/packages/grpc-web/test/generated_code_test.js b/packages/grpc-web/test/generated_code_test.js    
index de37cb0..3ae2070 100644                                                                                 
--- a/packages/grpc-web/test/generated_code_test.js                                                           
+++ b/packages/grpc-web/test/generated_code_test.js                                                           
@@ -362,7 +362,7 @@ describe('grpc-web generated code (commonjs+dts)', function() {                           
     [protoGenCodePath, protoDtsGenCodePath,                                                                  
      genCodePath, genDtsCodePath].map(f => {                                                                 
        if (fs.existsSync(f)) {                                                                               
-         fs.unlinkSync(f);                                                                                   
+         //fs.unlinkSync(f);                                                                                 
        }
      });
   });
```

Now you can run `./node_modules/.bin/tsc ./test/echo_grpc_web_pb.d.ts` repeatedly to reproduce the error.

The tests were last green on Feb 26. The first time this error is detected is Mar 4th. There is no significant code change that will affect this as far as I can tell between that period.

So what I see is that, after running `npm install` from the `packages/grpc-web` directory, this file `node_modules/grpc-web.d.ts` is being generated with one line in it

```
export * from '/usr/local/google/home/stanleycheung/grpc-web/packages/grpc-web'; 
```

The path corresponds correctly to the `index.d.ts` file on this package. My `package.json` file does have a line `"typings": "index.d.ts",`, and that had been there forever.

The [index.d.ts](https://github.com/grpc/grpc-web/blob/master/packages/grpc-web/index.d.ts) file itself hasn't been touched for a while. It seems fine to me with `declare module "grpc-web" {` as the beginning.


I tried to pin `tsc` back to an older version (moving back from the current 3.8 to say 3.2) and the error is still the same.

Any TypeScript expert can point me to the right direction? I had been googling this `TS2306` error and couldn't really figure out what's wrong. Any help will be highly appreciated!

--- a/packages/grpc-web/package.json
+++ b/packages/grpc-web/package.json
@@ -1,6 +1,6 @@
 {
   "name": "grpc-web",
-  "version": "1.0.7",
+  "version": "1.0.8",
   "author": "Google Inc.",
   "description": "gRPC-Web Client Runtime Library",
   "homepage": "https://grpc.io/",
@@ -30,7 +30,7 @@
     "gulp-eval": "^1.0.0",
     "mocha": "^5.2.0",
     "mock-xmlhttprequest": "^2.0.0",
-    "require-self": "^0.2.1",
-    "typescript": "^3.2.1"
+    "require-self": "0.2.1",
+    "typescript": "~3.7.0"
   }
 }

